core/locale/application/date_time_service_test.go

```diff
@@ -0,0 +1,133 @@
+package application
+
+import (
+	"flamingo.me/flamingo/v3/core/locale/domain"
+	"testing"
+	"time"
+
+	"flamingo.me/flamingo/v3/framework/flamingo"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDateTimeService_GetDateTimeFormatterFromIsoString(t *testing.T) {
+	start := time.Now()
+	now, err := time.Parse(time.RFC3339, start.Format(time.RFC3339))
+	assert.NoError(t, err, "no error received")
+
+	// error for invalid time
+	formatter, err := new(DateTimeService).GetDateTimeFormatterFromIsoString("invalid time")
+	assert.Nil(t, formatter, "got nil for formatter")
+	assert.Error(t, err, "error received")
+
+	// error for invalid location
+	dateTimeService := DateTimeService{}
+	dateTimeService.Inject(flamingo.NullLogger{}, &struct {
+		DateFormat     string `inject:"config:core.locale.date.dateFormat"`
+		TimeFormat     string `inject:"config:core.locale.date.timeFormat"`
+		DateTimeFormat string `inject:"config:core.locale.date.dateTimeFormat"`
+		Location       string `inject:"config:core.locale.date.location"`
+	}{
+		Location: "invalid location",
+	})
+
+	formatter, err = dateTimeService.GetDateTimeFormatterFromIsoString(now.Format(time.RFC3339))
+	assert.Nil(t, formatter, "got nil for formatter")
+	assert.Error(t, err, "error received")
+
+	// just get a plain formatter
+	plain := &domain.DateTimeFormatter{}
+	plain.SetDateTime(now, now.In(time.UTC))
+
+	formatter, err = new(DateTimeService).GetDateTimeFormatterFromIsoString(now.Format(time.RFC3339))
+	assert.Equal(t, plain, formatter, "got a formatter")
+	assert.NoError(t, err, "no error received")
+
+	// get a formatter for a configured locale
+	dateTimeService = DateTimeService{
+		logger: flamingo.NullLogger{},
+	}
+	dateTimeService.Inject(flamingo.NullLogger{}, &struct {
+		DateFormat     string `inject:"config:core.locale.date.dateFormat"`
+		TimeFormat     string `inject:"config:core.locale.date.timeFormat"`
+		DateTimeFormat string `inject:"config:core.locale.date.dateTimeFormat"`
+		Location       string `inject:"config:core.locale.date.location"`
+	}{
+		DateFormat:     time.RFC822,
+		TimeFormat:     time.Kitchen,
+		DateTimeFormat: time.ANSIC,
+		Location:       "America/New_York",
+	})
+
+	loc, err := time.LoadLocation("America/New_York")
+	assert.NoError(t, err, "no error received")
+
+	result := &domain.DateTimeFormatter{
```

Comment:
This is expected result - should be named like this.

In general I am not sure if we should test it that way (knowing how the method constructs the formatter and rebuild this in the test) in general - maybe we can test it more blackbox:
calling formatter.FormatDateTime and compare the string with what we expect?